fiftyone/core/odm/database.py

```diff
@@ -422,7 +422,7 @@ def get_async_db_conn(use_global=False):
 def _apply_options(db):
     timezone = fo.config.timezone
 
-    if not timezone or timezone.lower() == "utc":
+    if not timezone:
```

Comment:
If you specify `fo.config.timezone="utc"`, we should apply timezone awareness with `utc` timezone explicitly.
Otherwise the `datetime` coming from DB will have no timezone attached which is not the same as having UTC timezone. For example (see other file), `astimezone()` assumes local time in order to convert to UTC so changes the datetime value.